account_statement_import_online/models/online_bank_statement_provider.py

```diff
@@ -365,6 +363,14 @@ def _sanitize_bank_account_number(self, bank_account_number):
         self.ensure_one()
         return sanitize_account_number(bank_account_number)
 
+    def _update_partner_from_account_number(self, line_values):
+        partner_bank = self.env["res.partner.bank"].search(
+            [("acc_number", "=", line_values["account_number"])], limit=1
```

Comment:
No `company_id` restriction?

Reply:
I took this from the not online account_statement_import. res.partner.bank statements with company_id filled are own accounts, not partner accounts. Maybe I should add that company_id should be False.

Reply:
Can you have the same bank account number in several companies? If so, then you should check by company_id = company of the statement or company_id = False.

Reply:
O wait, that was in in old version of Odoo.

Reply:
Still, I do not see a need to check for company. You can have a partner without a company_id, meaning valid everywhere, and then when adding a bank_account, the res.partner.bank record gets the company_id of the logged in user. I think that is illogical, and unneeded, as it is still not a required field.